core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -206,6 +215,49 @@ private String encryptPassword(String password) {
         }
     }
 
+    /**
+     * Add tenant's public certificate to the database.
+     *
+     * @param keyStoreName Name of the key store.
+     * @param publicCert   Public certificate of the tenant.
+     * @throws SecurityException If an error occurs while adding the tenant's public certificate.
+     */
+    public void addTenantPublicKey(String keyStoreName, X509Certificate publicCert) throws SecurityException {
```

Comment:
This is used only in the tenant listener when generating the tenant's key-store.. So its not good to introduce just this public method for a such specific case.. Rather, check whether this can be handled by the `addKeystore()` method itself, when generating the tenant's primary key-store.

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/4099/commits/f89bbb95d862ca9f626a21592685bbd3e230eb11